polygon/heimdall/heimdall_no_store.go

```diff
@@ -0,0 +1,116 @@
+package heimdall
+
+import (
+	"context"
+
+	"github.com/ledgerwatch/log/v3"
+)
+
+//go:generate mockgen -destination=./heimdall_no_store_mock.go -package=heimdall . HeimdallNoStore
+type HeimdallNoStore interface {
```

Comment:
im a bit lost here... why do we need this?

Reply:
maybe we just need a `heimdall.Client` interface instead?

Reply:
or just add these "direct (not via DB/Store)" methods to the existing `heimdall.Heimdall` interface